### DISCUSSION TITLE
How to achieve region drawing of controls？

### DISCUSSION THREAD

**Discussion by ghost** [AvaloniaUI/Avalonia#17749] [Q&A]
![last](https://github.com/user-attachments/assets/f5092bec-0f23-477e-a0ed-1d398f9323a9)
![then](https://github.com/user-attachments/assets/baf094ab-68bd-4b00-8b78-1122b5f85bdc)
I drew all the images in the previous frame, and then in the next frame, I only want to update one area of the drawing. The other areas that have not been updated will keep the image drawn last time. However, now the other areas that have not been updated are showing a black screen. What methods can I use to achieve it?
![微信截图_20241211160144](https://github.com/user-attachments/assets/466a3c8e-adaf-4663-a419-83834a4be4d0)

**Comment by ghost** [comment]
This is just a demo. In fact, I want to display multiple camera images simultaneously on this custom control. When the latest frame arrives on one of the images, I only want to update the area where that camera is drawn, while keeping the rest unchanged.

**Comment by thevortexcloud** [comment]
What you are describing is called a blit. However it's generally done on the GPU. I am also not sure if Avalonia even exposes an API for it. That said you are not the first person to ask for this for this exact scenario:

https://github.com/AvaloniaUI/Avalonia/issues/16218

**Reply by ghost** [reply]
It doesn't seem like the solution I need, what I need is for the black screen area of the second image to maintain the content displayed in my previous image.
